diesel/src/pg/types/date_and_time/time.rs

```diff
@@ -17,6 +17,11 @@ use crate::sql_types::{Date, Time, Timestamp, Timestamptz};
 // Postgres timestamps start from January 1st 2000.
 const PG_EPOCH: PrimitiveDateTime = datetime!(2000-1-1 0:00:00);
 
+fn to_primitive_datetime(dt: OffsetDateTime) -> PrimitiveDateTime {
+    let dt = dt.to_offset(UtcOffset::UTC);
+    PrimitiveDateTime::new(dt.date(), dt.time())
```

Comment:
That seems arbitrary.
I think I would have expected the one that just drops timezone information (turns into naive) without converting to UTC.
But then there's really no non-arbitrary way to implicitly drop timezone information.
=> I would prefer that we do not do this

Reply:
It is arbitrary, but this is also just mirroring the current behavior of the existing `MySql` implementation.

If all your timestamps are sanitized internally to UTC it's reasonably common for applications to store without timezone to save space. Since local tz obviously doesn't work, the only reasonable assumption is UTC.

Reply:
The main difference to mysql here is that postgresql has a distinct timestamp type (`Timestamp with time zone`), which does exactly what you describe. So it seems to be a bit odd to reimplement that behavior here.

Given that there is no non-arbitrary way to implicitly drop timezone information to drop the timezone information I personally would prefer to keep that step explicit for the user. I would go as far and argue that accepting the mysql impl was a mistake back then. Maybe we should put a `#[deprecated]` on that impl instead?

Reply:
does trait implementation deprecation work now? 🤩

Reply:
I'm not sure if rustc will show the deprecation warning, but we have the `without-deprectaed` feature flag for such cases.